wellcomeml/ml/spacy_classifier.py

```diff
@@ -81,23 +83,24 @@ def fit(self, X, Y):
         X_train, X_test, Y_train, Y_test = train_test_split(
             X, Y, random_state=42
         )
+        # Free memory
+        del X
+        del Y
 
         nb_labels = Y_train.shape[1]
         self.unique_labels = [str(i) for i in range(nb_labels)]
         self._init_nlp()
         n_iter = self.n_iterations
-        train_texts = X_train
-        train_tags = self._label_binarizer_inverse_transform(Y_train)
-    
-        train_cats = [
-            {
-                label: label in tags
-                for label in self.unique_labels
-            } for tags in train_tags
-        ]
-
-        train_data = list(zip(train_texts, [{"cats": cats} for cats in train_cats]))
-
+        
+        def yield_train_data(X_train, Y_train):
+            for x, y in zip(X_train, Y_train):
+                tags = self._label_binarizer_inverse_transform([y])[0]
```

Comment:
so I take it that this will take longer transforming them one by one, but be better for memory? Is there an option to transform Y_train in chunks?

Reply:
yes there is `self._label_binarizer_inverse_transform(Y)` but as you said this would defeat the point. additionally the code within that function is not that optimised for matrices so it makes almost no difference whether you pass the entire matrix or examples one by one. in terms of memory though it helps as you do not have to save the transformation of Y in a separate variable.